examples/Forest_portability_example.ipynb

```diff
@@ -1,268 +1 @@
end` will allow us to view the full statevector (wavefunction) expected from a perfect execution of the circuit."]}, {"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["from pytket.extensions.pyquil import ForestStateBackend"]}, {"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["state_backend = ForestStateBackend()\n", "state_backend.compile_circuit(tk_circ)"]}, {"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["state = state_backend.get_state(tk_circ)\n", "print(state)"]}, {"cell_type": "markdown", "metadata": {}, "source": ["For users who are familiar with the Forest SDK, the association of qubits to indices of bitstrings (and consequently the ordering of statevectors) used by default in `pytket` Backends differs from that described in the [Forest docs](http://docs.rigetti.com/en/stable/wavefunction_simulator.html#multi-qubit-basis-enumeration). You can recover the ordering used by the Forest systems with `state_backend.get_state(tk_circ, basis:pytket.BasisOrder.dlo)` (see our docs on the `BasisOrder` enum for more details)."]}, {"cell_type": "markdown", "metadata": {}, "source": ["Connecting to real devices works very similarly. Instead of obtaining the full statevector, we are only able to measure the quantum state and sample from the resulting distribution. Beyond that, the process is pretty much the same.<br>\n", "<br>\n", "The following shows how to run the circuit on the \"9q-square\" lattice. The `simulator` switch on the `ForestBackend` will switch between connecting to the real Aspen device and the QVM, allowing you to test your code with a simulator before you reserve your slot with the device."]}, {"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["tk_circ.measure_all()"]}, {"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["from pytket.extensions.pyquil import ForestBackend"]}, {"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["aspen_backend = ForestBackend(\"9q-square\", simulator=True)\n", "aspen_backend.compile_circuit(tk_circ)"]}, {"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["counts = aspen_backend.get_counts(tk_circ, 2000)\n", "print(counts)"]}, {"cell_type": "markdown", "metadata": {}, "source": ["Note that attempting to connect to a live quantum device (using a `ForestBackend` with `as_simulator=False`) will fail unless it is running from a QMI instance during a reservation for the named lattice."]}], "metadata": {"kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "mimetype": "text/x-python", "name": "python", "nbconvert_exporter": "python", "pygments_lexer": "ipython3", "version": "3.6.4"}}, "nbformat": 4, "nbformat_minor": 2}
```

Comment:
I think here is a problem with the format of this file?

Reply:
What is the problem? It displays OK for me in a browser.
(Note these files are autogenerated, we don't control the formatting.)

Reply:
I thought it is maybe a bit hard to read without line breaks. Maybe black could split that up after the generation? If that is not possible just leave it as it is.

Reply:
Well these files aren't meant to be read by us: it is the Python files (in the `python` directory) that we will read and edit.
That was one reason to make this change: ipynb files are horrible to edit and diff at the best of times.

Reply:
Okay, thank you for the explanations. I am now fine with this.